src/main/java/org/jenkinsci/plugins/scriptsecurity/sandbox/whitelists/EnumeratingWhitelist.java

```diff
@@ -50,9 +50,7 @@
 
     protected abstract List<FieldSignature> staticFieldSignatures();
 
-    // TODO should precompute hash sets of signatures, assuming we document that the signatures may not change over the lifetime of the whitelist (or pass them in the constructor)
-
-    @Override public final boolean permitsMethod(Method method, Object receiver, Object[] args) {
+    @Override public boolean permitsMethod(Method method, Object receiver, Object[] args) {
```

Comment:
Making these non`final` smells like a bad approach. If nothing else, blocks future optimizations.

Reply:
It's not my favorite thing definitely, but this is what permits us to use the HashSet lookup. 

Unfortunately @jglick our hands are somewhat tied because `EnumeratingWhitelist` insists on returning a `List<SomeSignatureType>` for the various `blahSignatures` methods.  IMO that was a mistake -- if they took a Collection then we could internally rework the implementation perhaps to do something clever with `Set`s and then fall back to binary search or iteration, but that ship has sailed.